### PR TITLE
Code Review sample groovy script does not allow using custom rules

### PR DESCRIPTION
**Pull request by jbyibm**
See: https://github.com/IBM/dbb/issues/65

### CODE REVIEW

**Comment by jbyibm** [comment]
@dennis-behm  FYI

File: Pipeline/RunIDZCodeReview/RunCodeReview.groovy [hunk 1]
```diff
@@ -132,15 +132,15 @@ else
 		codeRev.getAllDDNames().each({ ddName ->
 			if (ddName == 'XML') {
 				def file = new File("${properties.workDir}/CodeReview${ddName}.xml")
-				codeRev.saveOutput(ddName, file, properties.logEncoding)
+				codeRev.saveOutput(ddName, file, properties.logEncoding/*, true*/)
```
Comment (dennis-behm) [inline Pipeline/RunIDZCodeReview/RunCodeReview.groovy:135]: @jbyibm
I think the new saveOutput method will break backward compatibility. While the new method will allow to trim and remove the ASA control characters from the JCL when saving the DD output, can we indicate the required DBB toolkit level?

File: Pipeline/RunIDZCodeReview/codereview.properties [hunk 2]
```diff
@@ -2,14 +2,15 @@
 
 #
 # Job card, please use \n to indicate a line break and use \ to break the line in this property file
-codereview_jobcard = //CODEREV JOB ,    		\n\
-// MSGCLASS=H,MSGLEVEL=(1,1),REGION=0M,COND=(16,LT) \n\
-//* 										\n\
-//PROCLIB  JCLLIB ORDER=DBEHM.#CUST.PROCLIB \n
```
Comment (dennis-behm) [inline Pipeline/RunIDZCodeReview/codereview.properties:8]: Thanks for the clean-up :-)

File: Pipeline/RunIDZCodeReview/codereview.properties [hunk 3]
```diff
@@ -21,7 +22,6 @@ codereview_includedFiles=**/*.cbl
 ## IDZ code review supports COBOL and PLI, Sample:codereview_includedFiles=**/*.cbl
```
Comment (dennis-behm) [inline Pipeline/RunIDZCodeReview/codereview.properties:22]: Can you just briefly update the comment for the variable.

--> This is used to identify the PDS libraries for the Copybooks based on the file extension of the include files

**Review by dennis-behm — CHANGES_REQUESTED** [review]
With the new method the JUNIT output file, code review results can be pushed to any dashboard, including the Jenkins build result view.

I am ok with breaking the backward compatibility - this is a sample.

We should indicate in the groovy file (probably also in the cli builder help) about the required DBB toolkit version.

File: Pipeline/RunIDZCodeReview/RunCodeReview.groovy (hunk 1 above)
Comment (jbyibm) [reply Pipeline/RunIDZCodeReview/RunCodeReview.groovy]: Updating DBB to the last PTF can be delayed in customer organizations. In my last commit I print this message if the new API is not available.
```
*? Warning the output file /var/jenkins/agent/genapp/workspace/GenAppNazarePipelineZunit_master/BUILD-211/build.20201217.032844.028/CodeReviewJUNIT.xml
*? will have an extra space at the beginning of each line.
*? Updating DBB to the latest PTF with ASA control characters API for JCLExec is highly recommended.
```

File: Pipeline/RunIDZCodeReview/codereview.properties (hunk 3 above)
Comment (jbyibm) [reply Pipeline/RunIDZCodeReview/codereview.properties]: Done

**Comment by jbyibm** [comment]
I also fix the situation where the rule files are too long to fit in a sigle JCL line.

**Review by dennis-behm — APPROVED** [review]
Merci!

**Pull request merged**